snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/tracker/Emitter.java

```diff
@@ -258,12 +270,19 @@ private Emitter(EmitterBuilder builder) {
         TLSArguments tlsArguments = new TLSArguments(this.tlsVersions);
         buildEmitterUri();
 
-        client = new OkHttpClient.Builder()
-                .sslSocketFactory(tlsArguments.getSslSocketFactory(),
-                        tlsArguments.getTrustManager())
-                .connectTimeout(15, TimeUnit.SECONDS)
-                .readTimeout(15, TimeUnit.SECONDS)
-                .build();
+        final OkHttpClient.Builder clientBuilder;
+        if (builder.client == null) {
+            clientBuilder = new OkHttpClient.Builder();
+        }
+        else {
+            clientBuilder = builder.client.newBuilder();
+        }
+
+        client = clientBuilder.sslSocketFactory(tlsArguments.getSslSocketFactory(),
+                                                tlsArguments.getTrustManager())
+                              .connectTimeout(15, TimeUnit.SECONDS)
+                              .readTimeout(15, TimeUnit.SECONDS)
+                              .build();
```

Comment:
I'm not really familiar with ok-http but if you already supplied a client, do you want to be recreating a builder? cc @ninjabear who has xp with ok-http

Reply:
According to the documentation in [1]

> Customize your client with newBuilder()
> You can customize a shared OkHttpClient instance with newBuilder(). This builds a client that shares the same connection pool, thread pools, and configuration. Use the builder methods to configure the derived client for a specific purpose.
> 
> This example shows a call with a short 500 millisecond timeout:
   > OkHttpClient eagerClient = client.newBuilder()
       .readTimeout(500, TimeUnit.MILLISECONDS)
       .build();
  > Response response = eagerClient.newCall(request).execute();
 
So I'm just creating a new builder to share the internals of the supplied client and just customize the needed parts. 

[1] https://square.github.io/okhttp/3.x/okhttp/okhttp3/OkHttpClient.html

Reply:
thanks for the explanation :+1: